src/test/csf/timers.h

```diff
@@ -0,0 +1,78 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#ifndef RIPPLE_TEST_CSF_TIMERS_H_INCLUDED
+#define RIPPLE_TEST_CSF_TIMERS_H_INCLUDED
+
+#include <chrono>
+#include <ostream>
+#include <test/csf/Scheduler.h>
+#include <test/csf/SimTime.h>
+
+namespace ripple {
+namespace test {
+namespace csf {
+
+/** Gives heartbeat of simulation to signal simulation progression
+ */
+using namespace std::chrono;
+class HeartbeatTimer
+{
+    Scheduler & scheduler_;
+    SimDuration interval_;
+
+    RealTime startRealTime_;
+    SimTime startSimTime_;
+
+public:
+    HeartbeatTimer(Scheduler& sched, SimDuration interval = 60s)
+            : scheduler_{sched}, interval_{interval},
+              startRealTime_{RealClock::now()},
+              startSimTime_{sched.now()}
+    {
+    };
+
+    inline void
+    start()
+    {
+        scheduler_.in(interval_, [this](){this->beat(scheduler_.now());});
```

Comment:
Correct, it would likely require another thread.  There is already https://github.com/ripple/rippled/blob/develop/src/ripple/core/DeadlineTimer.h and an example in https://github.com/ripple/rippled/blob/develop/src/test/core/DeadlineTimer_test.cpp.

Just food for thought; your approach above is useful as is.